File: alogrytm/figury.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  figury.py
#  




def choinka(h, znak):


    h = int(input("Podaj wysokość choinki: "))
    znak = str(input("Podaj znak, którym chcesz rysować: "))
    
    for i in range(h):
        for j in range(i + 1):
            print(znak, end = '')
        print()
        
    return 0
def choinka2(h, znak):


    h = int(input("Podaj wysokość choinki: "))
    znak = str(input("Podaj znak, którym chcesz rysować: "))
    
    for i in range(h):
        for j in range(h - i):
            print(znak, end = '')
        print()
        
    return 0
        
def prostokat1(a, b, znak):
    
    a = int(input("Podaj długość boku a: "))
    b = int(input("Podaj długość boku b: "))    
    znak = str(input("Podaj znak budujący prostokąt: "))    
    
    i = 0
    j = 0
    

    for i in range(a):
        for j in range(b):
            if j == 0 and j == b - 1:
                print(znak , end='')
            else:
                print("")
        
        print()
        



def tro(a, b, znak):
    a = int(input("Podaj długość boku a: "))
    b = int(input("Podaj długość boku b: "))    
    znak = str(input("Podaj znak budujący prostokąt: "))    
    
    i = 0
    j = 0
    

    for i in range(a):
        for j in range(b):
            print(znak , end='')
        print()
    
    
def trojkat(h, znak)
    
    h = int(input("Podaj wysokość choinki: "))
    znak = str(input("Podaj znak, którym chcesz rysować: "))
    
    for i in range(h):
        for j in range(h - i):
            print(znak, end = '')
        print()
        
    return 0
    
    
def main(args):
    h, znak = 6, "#"
    trojkat(h, znak)
    print()
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: alogrytm/figury.cpp
/*
 * figury.cpp
 */



#include <iostream>

using namespace std;

void prostokat(int x, int y, char z) 
{
    for (int i = 0; i < x; i++){
        for (int j = 0; j < y; j++)
            if (j == 0 || j == y-1 || i == 0 || i == x-1)
                cout << z;
            else
                cout << " ";
        cout << endl;
    }
}

int main(int argc, char **argv)
{
	int a, b;  //deklaracja
    a = b = 0;  //inicjacja
    
    cout << "Podaj długość boku a: ";
    cin >> a;
        
    cout << "Podaj długość boku b: ";
    cin >> b;
    
    char znak;
    
    cout << "Podaj znak, którym chcesz rysować: ";
    cin >> znak;
    
    prostokat(a, b, znak);
    
    /* 
     * cout << "Podaj długości boków prostokąta: ";
     * cin >> a >> b
     */
    
	return 0;
}
